### DISCUSSION TITLE
Python language evolution with Mojo

### DISCUSSION THREAD

**Discussion by jonathondduncan** [modular/modular#2426] [Q&A]
Hi all,

What happens if python begins to implement similar functionality and language features, such as using fn for strict typed functions, but in a different way than how mojo works now? Is there discussion between the python foundation and modular?

**Comment by geo-mak** [comment]
Regarding the goal of being a "superset" of Python in terms of syntax similarities, I can't say much. However, I expect it to be difficult to maintain, especially given that the two languages have totally different programming models, with different focus.

Other than that, Mojo executes standard Python code using CPython, which means that as long as your code is supported by CPython, it can be executed regardless of any changes made to the reference implementation of Python. Basically, you don't have to worry if you want to run your standard Python code, no conflict is going to happen.

**Comment by Benny-Nottonson** [comment]
Its an interesting question but I think its highly unlikely, that isn't even close to the roadmap for Python nor has it ever been mentioned. Along with that, I don't think that would be possible without a complete rewrite of CPython, which I also think is extremely unlikely.
